src/api/Controllers/UploadController.cs

```diff
@@ -0,0 +1,220 @@
+
+        csv.Context.RegisterClassMap(new BoreholeMap());
+
+        return csv.GetRecords<Borehole>().ToList();
+    }
+
+    private sealed class BoreholeMap : ClassMap<Borehole>
+    {
+        private readonly CultureInfo swissCulture = new("de-CH");
+
+        public BoreholeMap()
+        {
+            var config = new CsvConfiguration(swissCulture)
+            {
+                IgnoreReferences = true,
+                PrepareHeaderForMatch = args => args.Header.Humanize(LetterCasing.Title),
+            };
+
+            AutoMap(config);
+            Map(b => b.LocationX).Name("location_x_lv_95");
+            Map(b => b.LocationY).Name("location_y_lv_95");
+            Map(m => m.BoreholeCodelists).Convert(args =>
+            {
+                var boreholeCodelists = new List<BoreholeCodelist>();
+                new List<(string Name, int CodeListId)>
+                {
+                    ("id_geodin_shortname", 100000000),
+                    ("id_info_geol", 100000003),
+                    ("id_original", 100000004),
+                    ("id_canton", 100000005),
+                    ("id_geo_quat", 100000006),
+                    ("id_geo_mol", 100000007),
+                    ("id_geo_therm", 100000008),
+                    ("id_top_fels", 100000009),
+                    ("id_geodin", 100000010),
+                }.ForEach(id =>
+                {
+                    var value = args.Row.GetField<string>(id.Name);
+                    if (!string.IsNullOrEmpty(value))
+                    {
+                        boreholeCodelists.Add(new BoreholeCodelist
+                        {
+                            CodelistId = id.CodeListId,
+                            SchemaName = "borehole_identifier",
+                            Value = value,
+                        });
+                    }
+                });
+
+                return boreholeCodelists;
+            });
+            Map(b => b.Date).Name("date").Convert(args =>
+            {
+                string? dateString = args.Row.GetField<string>("date");
+                if (DateTime.TryParse(dateString, swissCulture, DateTimeStyles.None, out DateTime date))
+                {
+                    return date.ToUniversalTime();
+                }
+
+                return null;
+            });
+        }
+    }
+
+    private async Task<ActionResult<int>> SaveChangesAsync(Func<ActionResult<int>> successResult)
+    {
+        try
+        {
+            await context.UpdateChangeInformationAndSaveChangesAsync(HttpContext).ConfigureAwait(false);
+
+            return successResult();
+        }
+        catch (Exception ex)
+        {
+            var errorMessage = "An error occurred while saving the entity changes.";
+            logger?.LogError(ex, errorMessage);
+            return Problem(errorMessage, statusCode: StatusCodes.Status500InternalServerError);
+        }
+    }
+
+    private static bool HasData(Borehole borehole)
```

Comment:
PP: Diese Prüfung würde ich vorerst noch weglassen und dann zusammen mit den weiterführenden Validierungen im CsvHelper lösen. Ich nehme stark an, dass wir das mit dem CsvHelper lösen können bzw. inline in der Methode `ReadBoreholesFromCsv`.

Reply:
Ja das Problem ist, dass keins der Felder für den Import obligatorisch ist, also ist es derzeit mit "Bordmitteln" von csv Helper schwierig ist irgendwas zu validieren.
So ist die einzige Möglichkeit die ich gefunden habe um zu verhindern, dass wenn ich ein Random csv file mit x Reihen importiere, nicht einfach x leere Bohrungen angelegt werden.